authority/authority.go

```diff
@@ -615,3 +629,46 @@ func (a *Authority) requiresSCEPService() bool {
 func (a *Authority) GetSCEPService() *scep.Service {
 	return a.scepService
 }
+
+func (a *Authority) startCRLGenerator() error {
+
+	if a.config.CRL.CacheDuration.Duration <= 0 {
```

Comment:
You're already doing this check above, you can move the if conditions from above here.